crates/iceberg/src/expr/visitors/manifest_evaluator.rs

```diff
@@ -154,9 +155,19 @@ impl BoundPredicateVisitor for ManifestFilterVisitor<'_> {
         _predicate: &BoundPredicate,
     ) -> crate::Result<bool> {
         let field = self.field_summary_for_reference(reference);
+
         match &field.lower_bound {
-            Some(bound) if datum <= bound => ROWS_CANNOT_MATCH,
-            Some(_) => ROWS_MIGHT_MATCH,
+            Some(bound_bytes) => {
+                let bound = ManifestFilterVisitor::bytes_to_datum(
```

Comment:
@liurenjie1024 what's the cost of `datum_from_bytes` and `bytes_to_datum`? Do we need to introduce a new type in between?